sno/rich_tree_builder.py

```diff
@@ -0,0 +1,133 @@
+import contextlib
+
+import pygit2
+
+EMPTY_TREE_ID = '4b825dc642cb6eb9a060e54bf8d69288fbee4904'
```

Comment:
It is not accessible anywhere in pygit2 or libgit2, although they sometimes use it in tests.
Its just what the empty tree happens to hash to.

It could be a constant in pygit2. But to make it less magic, I've replaced `repo.get(EMPTY_TREE_ID)` with `repo.TreeBuilder().write()` which has the same result.